torch/distributed/pipelining/schedules.py

```diff
@@ -1749,6 +1749,26 @@ def _step_microbatches(
         self._update_losses(self._stages, losses)
 
 
+class _PipelineContext:
+    def __init__(
+        self,
+        schedule_ref: _PipelineSchedule,
+        arg_mbs: Optional[list[tuple]] = None,
+        kwarg_mbs: Optional[list[dict]] = None,
+        target_mbs: Optional[list] = None,
+        losses: Optional[list] = None,
+    ):
+        self.schedule_ref = schedule_ref
+        self.arg_mbs = arg_mbs
+        self.kwarg_mbs = kwarg_mbs
+        self.target_mbs = target_mbs
+        self.losses = losses
```

Comment:
Should we check the list length to ensure they are either None or match the length?